src/app/notebook/notebook-helpers.ts

```diff
@@ -0,0 +1,48 @@
+import { MimeBundle } from './notebook-types';
+
+// A list of the mime types we will display from a mime bundle, ordered from most to least "preferable".
+// To determine which item to display from a mime bundle, we'll select the item with the mime type that appears first in this list.
+export const supportedMimeTypes: string[] = ['image/png', 'image/webp', 'image/jpeg', 'image/gif', 'text/html', 'text/json', 'text/plain'];
+
+export function selectBestFromMimeBundle(mimeBundle: MimeBundle): { mimeType: string; data: string } {
```

Comment:
Not blocking for this, but generally, and there are exceptions, but the pattern is generally to create Angular services with corresponding spec files (usually we have the spec files :) ).